pkg/backup/volume_snapshot_action.go

```diff
@@ -59,6 +59,14 @@ func (a *volumeSnapshotAction) Execute(ctx ActionContext, volume map[string]inte
 
 	metadata := volume["metadata"].(map[string]interface{})
 	name := metadata["name"].(string)
+	labels := metadata["labels"]
+	var pvfailureDomainZone string
+	if labels != nil {
+		labelsMap := labels.(map[string]interface{})
+		pvfailureDomainZone = labelsMap["failure-domain.beta.kubernetes.io/zone"].(string)
+	} else {
+		ctx.log("Backup %q: PersistentVolume %q has no 'failure-domain.beta.kubernetes.io/zone' label", backupName, name)
```

Comment:
I think we probably need to return an error here, since we can't proceed if we don't know the zone.